scalahost-sbt/src/main/scala/scala/meta/scalahost/sbt/ScalahostSbtPlugin.scala

```diff
@@ -46,21 +59,75 @@ object ScalahostSbtPlugin extends AutoPlugin {
               )
             }
           List(
-            s"-Xplugin:${scalahostJar}",
+            s"-Xplugin:$scalahostJar",
             "-Yrangepos",
             "-Xplugin-require:scalahost"
           )
         case _ => Nil
       }
+    }
+  )
+
+  lazy val scalahostHasMirrorSettings: Seq[Def.Setting[_]] = Def.settings(
+    // Automatically set scalahostAggregate to projects listed in dependsOn(x % Scalahost)
+    scalahostAggregate := thisProject.value.dependencies.collect {
+      case x if x.configuration.exists(_ == Scalahost.name) => x.project
+    },
+    javaOptions ++= {
+      if (scalahostAggregate.value.isEmpty) Nil
+      else {
+        val sourcepath =
+          scalahostSourcepath.value
+            .map(_.getAbsolutePath)
+            .mkString(",")
+        val classpath =
+          scalahostClasspath.value
+            .flatMap(_.files.map(_.getAbsolutePath))
+            .mkString(java.io.File.pathSeparator)
+        List(
+          s"-Dscalameta.mirror.sourcepath=$sourcepath",
+          s"-Dscalameta.mirror.classpath=$classpath"
```

Comment:
How about we have multiple `-D` properties, as in `-D<projectname>.scalameta.sourcepath`, etc?